sql-plugin/src/main/scala/org/apache/spark/sql/rapids/ColumnarWriteStatsTracker.scala

```diff
@@ -52,10 +53,11 @@ trait ColumnarWriteTaskStatsTracker {
   /**
    * Process a new column batch to update the tracked statistics accordingly.
    * The batch will be written to the most recently witnessed file (via `newFile`).
+   *
    * @param filePath Path of the file which the batch is written to.
-   * @param batch Current data batch to be processed.
+   * @param spillableBatch Current spillable data batch to be processed.
    */
-  def newBatch(filePath: String, batch: ColumnarBatch): Unit
+  def newBatch(filePath: String, spillableBatch: SpillableColumnarBatch): Unit
```

Comment:
Not sure about this change.  At this point we're actively trying to wield the batch, (we're in the middle of trying to write it), and thus it shouldn't be mandatory that it be spillable at this point, IMO.  Also would like to keep the tracker API using pure Spark classes to mirror the columnar form of regular job/task stat trackers.  If a particularly stat tracker does something complex that requires spill support (e.g.: Delta Lake stat gathering in their tracker), then that tracker can make the batch spillable when that makes sense.